packages/components/src/organisms/SKUMatrix/provider/SKUMatrixProvider.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { createContext, useCallback, useState } from 'react'
+import type { ReactNode } from 'react'
+
+interface IAllVariantProducts {
+  id: string
+  name: string
+  image: {
+    url: string
+    alternateName: string
+  }
+  inventory: number
+  availability: string
+  price: number
+  listPrice: number
+  priceWithTaxes: number
+  listPriceWithTaxes: number
+  specifications: Record<string, string>
+  selectedCount: number
+  offers: {
+    highPrice: number
+    lowPrice: number
+    lowPriceWithTaxes: number
+    offerCount: number
+    priceCurrency: string
+    offers: Array<{
+      listPrice: number
+      listPriceWithTaxes: number
+      sellingPrice: number
+      priceCurrency: string
+      price: number
+      priceWithTaxes: number
+      priceValidUntil: string
+      itemCondition: string
+      availability: string
+      quantity: number
+    }>
+  }
+}
+
+export interface SKUMatrixProviderContextValue {
+  /*
+    value that indicates if the modal is open
+  */
+  isOpen: boolean
+  /*
+    array of all variant products
+  */
+  allVariantProducts: IAllVariantProducts[]
+  /*
+    function to set the array of all variant products
+  */
+  setAllVariantProducts(
+    items: React.SetStateAction<IAllVariantProducts[]>
+  ): void
+  /*
+  */
+  handleChangeQuantityItem(id: string, value: number): IAllVariantProducts[]
+  /*
+    function to set the modal is open
+  */
+  setIsOpen(value: boolean): void
+}
+
+export const SKUMatrixContext =
+  createContext<SKUMatrixProviderContextValue | null>(null)
+
+function SKUMatrixProvider({ children }: { children: ReactNode }) {
+  const [isOpen, setIsOpen] = useState<boolean>(false)
+  const [allVariantProducts, setAllVariantProducts] = useState<
+    IAllVariantProducts[]
+  >([])
+
+  const handleChangeQuantityItem = useCallback(
+    (id: string, value: number) => {
+      const data = [...allVariantProducts]
+      const findedSKU = data.find((item) => item.id === id)!
+      findedSKU.selectedCount = value
+
+      setAllVariantProducts(data)
+
+      return data
+    },
+    [allVariantProducts]
+  )
```

Comment:
I think we can simplify and avoid the mutability of objects here. What about:

```suggestion
  const handleChangeQuantityItem = useCallback((id: string, value: number) => {
    setAllVariantProducts((prev) =>
      prev.map((item) =>
        item.id === id ? { ...item, selectedCount: value } : item
      )
    )
  }, [])
```
why? [reference](https://react.dev/learn/updating-arrays-in-state).